crates/compiler/builtins/roc/Num.roc

```diff
@@ -661,6 +662,16 @@ isLte : Num a, Num a -> Bool
 ## is [defined to be unordered](https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN).)
 isGte : Num a, Num a -> Bool
 
+## Returns `Bool.true` if the first number and second number are within a specific threshold
+##
+## A specific relative and absolute tolerance can be provided to change the threshold
+##
+## If either argument is [*NaN*](Num.isNaN), returns `Bool.false` no matter what. (*NaN*
+## is [defined to be unordered](https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN).)
+isApproxEq : Frac a, Frac a, { relativeTolerance ? Frac a, absoluteTolerance ? Frac a } -> Bool
+isApproxEq = \a, b, { relativeTolerance? 0.00001, absoluteTolerance? 0.000000001 } 
+    -> Num.abs (a-b) <= absoluteTolerance + relativeTolerance * Num.abs b
```

Comment:
How does this handle infinities and denormals?
I found this to be a good resource in the past: https://floating-point-gui.de/errors/comparison/

Reply:
Excellent question.  Shortcut of a == b returning true will be useful for infinity.

I will add all the test cases described in the link including:
Infinity
-Infinity
Nan
MinValue
MaxValue
etc.

Reply:
A couple of thoughts:

1. Do we want [`absDiff`](https://www.roc-lang.org/builtins/Num#absDiff) here?
2. In general `-` and `*` can crash on overflow. It feels like an equality check should never crash though, right? Should these use saturating addition and multiplication maybe?
3. [`abs`](https://www.roc-lang.org/builtins/Num#abs) can also crash for integers (e.g. `Num.abs Num.minI64` crashes, and it's not fixable), which I *think* means it can crash on `Dec` (cc @bhansconnect - does that sound right?) - we can't test this yet because we don't have `Num.abs` implemented for `Dec` yet though.

Reply:
> Do we want [absDiff](https://www.roc-lang.org/builtins/Num#absDiff) here?

Yeah, probably.

> In general - and * can crash on overflow. It feels like an equality check should never crash though, right? Should these use saturating addition and multiplication maybe?

That shouldn't be the case for floats. I guess for Dec we may have a problem. Oh, though the multiplication will always make the value smaller, so no overflow possible.

> [abs](https://www.roc-lang.org/builtins/Num#abs) can also crash for dec

Yeah, technically...Though if you have the absolute most negative dec, you probably are gonna crash anyway. That is very unlikely. It is literally part way through a number. Much more likely with the various ints, so I wouldn't be too worried about that.